### PR TITLE
Jetpack focus show or hide automattic family section on about screen

### PR DESCRIPTION
**Pull request by ravishanker**
This PR adds config for showing or hiding the following
- About Automattic Family row
- About Footer (Automattic logo)

Depends on corresponding [about-automattic-library PR](https://github.com/Automattic/about-automattic-android/pull/39)

| WordPress | Jetpack |
|---|---|
|Screenshot_20221006_164935|Screenshot_20221006_170517|

To test:

Test 1:
- Build and Run WPAndroid 
- Go to Me > About WordPress
- Ensure About Automattic Family row is not shown 
- Ensure automattic logo is not shown at the bottom

Test 2:
- Build and Run JPAndroid 
- Go to Me > About Jetpack
- Ensure About Automattic Family row is shown
- Ensure About Automattic Family doesn't have WordPress bubble in it
- Ensure automattic logo is shown at the bottom

Test 3:
- Check WP twitter link goes to [@WPAndroid](https://twitter.com/wpandroid)
- Check JP twitter link goes to [@jetpack](https://twitter.com/jetpack)

Test 4:
- Check WP blog title is **News** and points to https://wordpress.org/news/ 
- Check JP blog title is Blog and points to https://jetpack.com/blog/

Test 5:
- Check WP Contribute is present and points to https://make.wordpress.org/mobile/handbook
- Check JP has Work With Us and points to https://automattic.com/work-with-us

NOTE:
- Terms of service and Privacy policy links will be updated later along with  links on Play store

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested both WPAndroid and Jetpack variants

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻
I suggested a change on About Automattic library, but it's only a document change. So I'm approving this but not merging now.

**Comment by guarani** [comment]
@ravishanker, I think the PR title "Jetpack focus show or hide automattic family section on about screen" doesn't capture the scope of the changes, which also includes things mentioned in tests 3 to 5 above.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 21.0
 -----
+* [***] Updated About screen for WordPress and Jetpack apps to their respective urls. [https://github.com/wordpress-mobile/WordPress-Android/pull/17282]
```
Comment (AliSoftware) [inline RELEASE-NOTES.txt:5]: @ravishanker Any reason why this was marked with `[***]`, suggesting it to be a very important and impacting user-facing change?

To me, this change probably doesn't warrant being the most prominent item we want to highlight in our Release Notes (especially for Jetpack where the visible change will be even more minimal), and instead should probably be using the same priority indicator level as the entry for the Splash Screen update that is mentioned just below? Wdyt?

Comment (ravishanker) [reply RELEASE-NOTES.txt:5]: Sounds good.  It should be at the same level as Splash Screen.  Thank you.

**Pull request merged**